### PR TITLE
[10.0][MIG] web_readonly_bypass : Migration to 10.0

### PR DESCRIPTION
**Pull request by Ehtaga**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
There's already another migration in https://github.com/OCA/web/pull/451. Have you checked it?

**Comment by Ehtaga** [comment]
@pedrobaeza Sorry I didn't see this one. I saw #601 01 and it was not working so I started from the beginning. I can add the 2 other authors if you want to (in the README, not in the manifest as they did...).

**Comment by pedrobaeza** [comment]
If their contribution is valuable for this PR, then add them on contributors list. If it's not working, then we simply dispose that PR. Have you seen also https://github.com/OCA/web/issues/505?

**Comment by Ehtaga** [comment]
@pedrobaeza I added one of the contributor who did a similar migration.
I also cherry picked 2 commits from the 8.0 version.

**Comment by Ehtaga** [comment]
@pedrobaeza We noticed a bug with this module in v10. It works, but we can't load the Settings dashboard anymore.
The error is: `odoo could not find client action 'web_settings_dashboard.main'.`
For the moment we choose not to use this module as our project needed to be deployed quickly. If we have some time later to investigate, we will do it.
Can you add the 'need fixing' tag?
This PR is the most advanced among those you referred to.
It could also be interesting to look at the stylistic guidelines, as you suggested in #451 .
Thanks

File: web_readonly_bypass/static/src/js/readonly_bypass.js [hunk 1]
```diff
@@ -138,4 +138,47 @@
         },
 
     });
+
+    instance.web.ProxyDataSet.include({
```
Comment (yajo) [inline web_readonly_bypass/static/src/js/readonly_bypass.js:142]: You have to migrate the whole file to the new module-based JS API.

**Review by yajo — CHANGES_REQUESTED** [review]

**Comment by JayVora-SerpentCS** [comment]
@Ehtaga and @blutecsolutions need to work this out together as #601 was PRed earlier.

File: web_readonly_bypass/static/src/js/readonly_bypass.js (hunk 1 above)
Comment (anhvu-sg) [reply web_readonly_bypass/static/src/js/readonly_bypass.js:142]: Hello, do we have any update?

Comment (simahawk) [reply web_readonly_bypass/static/src/js/readonly_bypass.js:142]: JS has been ported here https://github.com/OCA/web/pull/743

Comment (yajo) [reply web_readonly_bypass/static/src/js/readonly_bypass.js:142]: Can we close this one then?

**Pull request closed without merging**